CRM/Contact/Form/Task/Map/Event.php

```diff
@@ -50,6 +50,13 @@ public function preProcess() {
     self::createMapXML($ids, $lid, $this, TRUE, $type);
     $this->assign('single', FALSE);
     $this->assign('skipLocationType', TRUE);
+
+    $today = date("Y-m-d G:i:s");
+    $startDate = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_Event', $ids, 'start_date');
+    $is_public = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_Event', $ids, 'is_public');
+    if ($is_public == 0 OR $startDate < $today) {
+      CRM_Utils_System::addHTMLHead('<META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW">');
```

Comment:
Can we create a function addNoFollowHead or similar - I feel like we might get a situation where the preferrred meta for no-follow changes & it would be good to only change in one place.

Reply:
+1 for this idea. Grepping shows there are 4 other spots in the codebase already containing this line, so refactoring them into a function makes sense.